### ISSUE TITLE
Kubernetes iSCSI Volume Support

### ISSUE THREAD

**Issue by fti7** [rancher/rancher#9685]
See https://forums.rancher.com/t/cant-mount-iscsi-volume/5413
Is there anything planned?

BR,
Frank

**Comment by polivbr** [comment]
Any updates on this? The rancher/netes image needs to be built with the open-iscsi package installed.

**Comment by innergy** [comment]
I think this and #9885 are duplicates. This is a significant barrier to adoption for many people wanting to use PVs. We have customers that want to use Rancher with k8s that can't due to this issue, and other k8s distributions do not share this limitation.

I believe the problem stems from the fact that your kubelet is containerized and doesn't have access to the iSCSI tools on the host even if they are installed there.

In other words, mounting iSCSI volumes from the host works fine, but the kubelet is unable to do it the way that it expects.

Can we crank up the priority of this? Thanks!

**Comment by davisvansant** [comment]
It seems we've been hit with this as well, and I think the solution is fairly simple, if I'm understanding things correctly

The basic rundown here is the following (note that for testing, this is all currently running on a single node):

- Rancher
- Kubernetes
- SolidFire/Trident on bare metal

After a quick "hack" to the container that runs the kubelet service, everything works as expected and is flawless

The "hack" is to exec into the container running the kubelet service, and run the following

```
apt-get update
apt-get install open-iscsi -y
```
I've verified this a few times, by setting up the environment from scratch, and with the initial pull of the Rancher image that holds the kubelet binaries, this doesnt work, but immediately when I exec into the container and run whats mentioned above, it works.

I've also verified locally that the latest image (https://hub.docker.com/r/rancher/k8s/) doesnt seem to have the `open-iscsi` packages, which, for reference, I believe is required to mount the iscsi volume(s) using the `iscsiadm` utility 

The workaround for our use case is simple, but is not scalable and we will be hesitant to move forward with this into production until this is resolved, unless we can do something like pull our own customized image with the tools installed (less than ideal)

So I think that to resolve this we may just be able to add the `open-iscsi` packages to the following Dockerfile (or parent, `rancher/agent-base:v0.3.0`, depending)

 - https://github.com/rancher/kubernetes-package/blob/master/package/Dockerfile#L3

I'd be glad to offer any further help and troubleshooting around this - thanks for all your help here!

**Comment by tacid** [comment]
We are facing the same problem here, there is no open-iscsi package installed inside kubelet service image. Not such a big deal to add this package inside an image.

**Comment by xetys** [comment]
this is also important to make OpenEBS running on rancher OOTB. Related issue: https://github.com/openebs/openebs/issues/1020

**Comment by deniseschannon** [comment]
There were no regressions introducing this package into the kubelet service. 

Available in `k8s:v1.8.5-rancher2`
